src/common/dao/webhook_job.go

```diff
@@ -0,0 +1,127 @@
+package dao
+
+import (
+	"fmt"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+)
+
+// UpdateWebhookJob update webhook job
+func UpdateWebhookJob(job *models.WebhookJob, props ...string) (int64, error) {
+	if job.ID == 0 {
+		return 0, fmt.Errorf("webhook job ID is empty")
+	}
+
+	o := GetOrmer()
+	return o.Update(job, props...)
+}
+
+// UpdateWebhookJobStatus ...
+func UpdateWebhookJobStatus(id int64, status string, statusCondition ...string) (int64, error) {
+	qs := GetOrmer().QueryTable(&models.WebhookJob{}).Filter("id", id)
+	if len(statusCondition) > 0 {
+		qs = qs.Filter("status", statusCondition[0])
+	}
+	params := orm.Params{
+		"status": status,
+	}
+
+	n, err := qs.Update(params)
+	if err != nil {
+		return 0, err
+	}
+	log.Debugf("update webhook job status %d: -> %s", id, status)
+	return n, err
+}
+
+// AddWebhookJob insert new webhook job to DB
+func AddWebhookJob(job *models.WebhookJob) (int64, error) {
+	o := GetOrmer()
+	if len(job.Status) == 0 {
+		job.Status = models.JobPending
+	}
+	return o.Insert(job)
+}
+
+// GetWebhookJob ...
+func GetWebhookJob(id int64) (*models.WebhookJob, error) {
+	o := GetOrmer()
+	j := &models.WebhookJob{
+		ID: id,
+	}
+	err := o.Read(j)
+	if err == orm.ErrNoRows {
+		return nil, nil
+	}
+	return j, nil
+}
+
+// GetTotalCountOfWebhookJobs ...
+func GetTotalCountOfWebhookJobs(query ...*models.WebhookJobQuery) (int64, error) {
+	qs := webhookJobQueryConditions(query...)
+	return qs.Count()
+}
+
+// GetWebhookJobs ...
+func GetWebhookJobs(query ...*models.WebhookJobQuery) ([]*models.WebhookJob, error) {
+	var jobs []*models.WebhookJob
+
+	qs := webhookJobQueryConditions(query...)
+	if len(query) > 0 && query[0] != nil {
+		qs = paginateForQuerySetter(qs, query[0].Page, query[0].Size)
+	}
+
+	qs = qs.OrderBy("-UpdateTime")
+
+	_, err := qs.All(&jobs)
+	return jobs, err
+}
+
+// GetLastTriggerInfosGroupByHookType get webhook jobs info of policy, including hook type and last trigger time
+func GetLastTriggerInfosGroupByHookType(policyID int64) ([]*models.LastTriggerInfo, error) {
```

Comment:
Such implementation the status of the job is lost.

Reply:
> Such implementation the status of the job is lost.

This method is designed for UI displaying. if status is wanted, we cloud use method `GetWebhookJobs` or `GetWebhookJob`